lib/chat_api/conversations.ex

```diff
@@ -496,6 +508,57 @@ defmodule ChatApi.Conversations do
     |> Repo.update()
   end
 
+  @spec find_or_create_customer_and_conversation(String.t(), String.t()) ::
+          {:ok, Customer.t(), Conversation.t()}
+          | {:error, Ecto.Changeset.t()}
+  def find_or_create_customer_and_conversation(account_id, phone) do
+    now = DateTime.utc_now()
+    sms_source = "sms"
+
+    case Customers.list_customers(account_id, %{phone: phone}) do
```

Comment:
If if I create a `Customers.find_or_create_by_phone`, It will always return a customer. 

The current code right contains this logic
```
if the customer is not present
    creates a customer + conversation
    return {customer, conversation}
else 
  find conversation
    if no conversation found
        create one
   else
        return {customer, conversation}
```
 If the customer is not present it creates the customer + conversation right away, without having to do a `find` by conversation

Reply:
hmm so wouldn't it still work if `Customers.find_or_create_by_phone` just indicated whether the returned customer was `found` or `created`?

maybe something like...

```elixir
case Customers.find_or_create_by_phone(account, phone) do
  {:ok, :found, customer} -> 
    case Conversations.find_latest_conversation_by_customer(customer.id) do
      nil -> Conversations.create_conversation(...)
      conversation -> {:ok, conversation}
    end

  {:ok, :created, customer} -> 
    Conversations.create_conversation(...)

  error -> 
    error
end
```

does that make sense?

Reply:
Yeah that makes sense!